SingularityUI/app/components/machines/Slaves.jsx

```diff
@@ -71,7 +72,48 @@ const Slaves = (props) => {
 
   const expiringSlaveState = (slave) => (
     _.find(props.expiringSlaveStates, (expiring) => expiring.machineId == slave.id)
-  )
+  );
+
+  const getMaybeDeactivateHostButton = (slave) => (
+    ! Utils.isIn(slave.host, props.inactiveHosts) && (
+      <FormModalButton
+        name="Mark Inactive"
+        buttonChildren={<Glyphicon glyph="remove-circle" />}
+        action="Mark Host Inactive"
+        onConfirm={() => props.deactivateHost(slave.host)}
+        tooltipText={`Flag host '${slave.host}' as inactive`}
+        formElements={[]}
+      >
+        <p>Are you sure you want to mark the host {slave.host} as inactive?</p>
+        <p>
+          This will decommission every slave on this host until you reactivate
+          it. You will also need to manually re-commission the slaves from
+          this host after you reactivate it.
+        </p>
+      </FormModalButton>
+    )
+  );
+
+  const getMaybeReactivateHostButton = (slave) => (
+    Utils.isIn(slave.host, props.inactiveHosts) && (
+      <FormModalButton
+        name="Reactivate Host"
+        buttonChildren={<Glyphicon glyph="ok-circle" />}
+        action="Mark Host Active"
+        onConfirm={() => props.reactivateHost(slave.host)}
+        tooltipText={`Mark host '${slave.host}' as active`}
+        formElements={[]}
+      >
+        <p>Are you sure you want to reactivate host {slave.host}?</p>
+        <p>
+          This will allow new slaves from this host to make offers without
```

Comment:
```
New slaves with this hostname will no longer be automatically marked as decommissioned
```

The making offers piece is a bit misleading. For one, we get the offers form the master, not the slave. Also, we will continue to get offers from the master for that slave as long as it is still connected. From Singularity's perspective, the important part is that it is a new slave, not necessarily that it continues to make offers